pom.xml

```diff
@@ -70,6 +71,16 @@
       <artifactId>junit</artifactId>
       <scope>provided</scope>
       <optional>true</optional>
+      <exclusions>
+        <!--
+        This has to be excluded because it's subjecting the code to the
+        incorrect version of certain hamcrest packages.
+        -->
+        <exclusion>
+          <groupId>org.hamcrest</groupId>
+          <artifactId>hamcrest-core</artifactId>
```

Comment:
Nice catch! I would also put an additional `<exclude>org.hamcrest:hamcrest-core</exclude>` line into the `maven-enforcer-plugin` configuration to avoid the legacy hamcrest artifact

Reply:
I can add that.